src/JsonSchema/TypeFactory.php

```diff
@@ -117,14 +117,16 @@ private function getClassType(?string $className, bool $nullable, string $format
             ];
         }
         if (!$this->isResourceClass($className) && is_a($className, \BackedEnum::class, true)) {
-            $rEnum = new \ReflectionEnum($className);
-            $enumCases = array_map(static fn (\ReflectionEnumBackedCase $rCase) => $rCase->getBackingValue(), $rEnum->getCases());
+            $enumCases = array_map(static fn(\BackedEnum $enum): string|int => $enum->value, $className::cases());
+
+            $type = is_string($enumCases[0] ?? '') ? 'string' : 'int';
```

Comment:
although it is not forbidden in PHP, it would be very weird to have an enum with 0 cases... not sure we should do something special in this case

Reply:
Can't we use `bool` or any kind of type for that matter?